Documentation/building/unix-instructions.md

```diff
@@ -0,0 +1,75 @@
+Building WCF on Linux
+==========================================
+
+WCF can be built on top of current [Mono CI builds](#installing-mono-packages) or a direct [build/install of Mono](http://www.mono-project.com/docs/compiling-mono/). It builds using MSBuild and Roslyn and requires changes that have not yet made it to official released builds.
+
+After preparing Mono, clone if you haven't already, and run the build script.
+
+```
+git clone https://github.com/dotnet/wcf.git
+cd wcf
+./build.sh
+```
+
+>These instructions have been validated on:
+* Ubuntu 14.04  
+
+>They may work on other unix operating systems but are untested
+
+
+# Installing Mono Packages
+_Mono installation instructions are taken from ["Install Mono"](http://www.mono-project.com/docs/getting-started/install/) and ["Continuous Integration Packages"](http://www.mono-project.com/docs/getting-started/install/linux/ci-packages/)._
+
+_**Note:** CI packages are not produced for Mac. As WCF needs current bits you must build Mono [yourself](http://www.mono-project.com/docs/compiling-mono/)._
+### Add Mono key and package sources
+##### Debian/Ubuntu (and other derivatives)
+```
+sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+echo "deb http://download.mono-project.com/repo/debian wheezy main" | sudo tee /etc/apt/sources.list.d/mono-xamarin.list
+echo "deb http://jenkins.mono-project.com/repo/debian sid main" | sudo tee /etc/apt/sources.list.d/mono-jenkins.list
+sudo apt-get update
+```
+##### Fedora/CentOS (and other derivatives)
+```
+sudo rpm --import "http://keyserver.ubuntu.com/pks/lookup?op=get&search=0x3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF"
+sudo yum-config-manager --add-repo http://download.mono-project.com/repo/centos/
+sudo yum-config-manager --add-repo http://jenkins.mono-project.com/repo/centos/
+sudo yum upgrade
+``` 
+### Install CI build and reference assemblies
+Install a recent (Continuous Integration) Mono build and the PCL reference assemblies. (_This instruction installs latest. To see available Mono builds, use `apt-cache search mono-snapshot` (Ubuntu) or `yum search mono-snapshot` (Fedora)_)
+##### Debian/Ubuntu (and other derivatives)
+```
+sudo apt-get install mono-snapshot-latest referenceassemblies-pcl
+```
+##### Fedora/CentOS (and other derivatives)
+```
+sudo yum install mono-snapshot-latest referenceassemblies-pcl
+```
+### Switch to the mono snapshot build
+```
+. mono-snapshot mono
+```
+
+# Known Issues
+If you see errors along the lines of `SendFailure (Error writing headers)` you may need to import trusted root certificates:
+```
+mozroots --import --sync
```

Comment:
I have no idea. I copied this from CoreFx. I created a second VM to verify the instructions and didn't run that command and everything built fine. I left it in as the only root cert stuff done that I'm aware of is downloading from nuget/myget and we do that too. It could be an issue with a different unix variation.